infra/build/functions/trial_build.py

```diff
@@ -273,30 +270,64 @@ def wait_on_builds(build_ids, credentials, cloud_project):
 
   wait_builds = build_ids.copy()
   build_results = {}
+  failed_builds = {}
+  total_builds = len(wait_builds)
+  next_check_time = datetime.datetime.now() + datetime.timedelta(hours=1)
+  notify_time = end_time - datetime.timedelta(minutes=15)
+  notified_timeout = False
+  logging.info(
+      '----------------------------Build result----------------------------')
+  logging.info(f'Trial build end time: {end_time}')
+  logging.info(f'Total builds: {total_builds}, {wait_builds}')
+  logging.info('Failed project, Statuses, Logs')
   while wait_builds:
-    logging.info('Polling: %s', wait_builds)
+    current_time = datetime.datetime.now()
+    # Update status every hour
+    if current_time >= next_check_time:
+      logging.info(
+          f'[{current_time}] Remaining builds: {len(wait_builds)}, {wait_builds}'
+      )
+      next_check_time += datetime.timedelta(hours=1)
+
+    # Notify if the build times out in the next 15 minutes.
+    if not notified_timeout and current_time >= notify_time:
+      notified_timeout = True
+      logging.info(
+          f'[{current_time}] Warning: trial build may time out in 15 minutes.\n'
+          f'Remaining builds: {len(wait_builds)}/{total_builds}, {wait_builds}.\n'
+          f'Failed builds: {len(failed_builds)}/{total_builds}, {failed_builds}'
+      )
+
     for project, project_build_ids in list(wait_builds.items()):
       for build_id in project_build_ids[:]:
         if check_finished(build_id, project, cloudbuild_api, cloud_project,
                           build_results):
+          if build_results[project] != 'SUCCESS':
+            logs_url = build_lib.get_logs_url(build_id)
+            failed_builds[project] = logs_url
+            logging.info(f'{project}, {build_results[project]}, {logs_url}')
 
           wait_builds[project].remove(build_id)
           if not wait_builds[project]:
             del wait_builds[project]
 
         time.sleep(1)  # Avoid rate limiting.
 
-  print('Printing results')
-  print('Project, Statuses')
-  for project, build_result in build_results.items():
-    print(project, build_result)
+  if not failed_builds:
+    logging.info(f'Summary: trial build passed.')
+  else:
+    logging.info(
+        f'Summary: trial build failed\n'
+        f'Failed builds: {len(failed_builds)}/{total_builds}, {failed_builds}')
 
   # Return failure if nothing is built.
   return all(build_results.values()) if build_results else False
```

Comment:
Since we change the `build_result` format, I think we need to change these to

```python
all(result == 'SUCCESS' for result in build_results) 
```

?